plugin/goto_diagnostic.py

```diff
@@ -77,18 +80,61 @@ def input_description(self) -> str:
         return "Goto Diagnostic"
 
 
-class DiagnosticUriInputHandler(sublime_plugin.ListInputHandler):
+ListItemsReturn = Union[List[str], Tuple[List[str], int], List[Tuple[str, Any]], Tuple[List[Tuple[str, Any]], int],
+                        List[sublime.ListInputItem], Tuple[List[sublime.ListInputItem], int]]
+
+
+class PreselectedListInputHandler(sublime_plugin.ListInputHandler, metaclass=ABCMeta):
+    """
+    Similar to ListInputHandler, but allows to preselect a value like some of the input overlays in Sublime Merge.
+    Inspired by https://github.com/sublimehq/sublime_text/issues/5507.
+
+    Subclasses of PreselectedListInputHandler must not implement the `list_items` method, but instead `_list_items`,
+    i.e. just prepend an underscore to the regular `list_items`.
+
+    When an instance of PreselectedListInputHandler is created, it must be given the window as an argument.
+    An optional second argument `initial_value` can be provided to preselect a value.
+    """
+
+    _window = None  # type: Optional[sublime.Window]
+    _initial_value = None  # type: Optional[Union[str, sublime.ListInputItem]]
+    _preselect = False
+
+    def __init__(
+        self, window: sublime.Window, initial_value: Optional[Union[str, sublime.ListInputItem]] = None
+    ) -> None:
+        super().__init__()
+        if initial_value is not None:
+            self._window = window
+            self._initial_value = initial_value
+            self._preselect = True
+
+    def list_items(self) -> ListItemsReturn:
+        if self._preselect and self._initial_value is not None and self._window is not None:
+            self._preselect = False
+            sublime.set_timeout(functools.partial(self._window.run_command, 'select'))
```

Comment:
Could/should we drop `_preselect` completely and just reset the `self._initial_value` to None? We'd just add a method here that selects and resets it:

```py
    def _select_and_reset_initial_value(self) -> None:
        if self._window:
            self._window.run_command('select')
            self._initial_value = None
```

Reply:
I originally had the `sublime.set_timeout` logic within `__init__` and moved it into `list_items` later. I think that was the reason why I would need another variable `_preselect`, but now it's not needed anymore and your suggestion works fine too. 👍